boards/arm/warp7_m4/doc/warp7_m4.rst

```diff
@@ -0,0 +1,341 @@
+expose the eMMC as mass storage with the following command in the u-boot
+prompt: ``ums 0 mmc 0``. Copy the compiled zephyr.bin to the first FAT
+partition and remove the mounted device on the PC by issuing a "Ctrl+C" in the
+u-boot prompt.
+Set the u-boot environment variables and run the zephyr.bin from the
+appropriated memory configured in the Zephyr compilation:
+
+.. code-block:: console
+
+   setenv bootm4 'fatload mmc 0:1 $m4addr $m4fw && dcache flush && bootaux $m4addr'
+   # TCML
+   setenv m4tcml 'setenv m4fw zephyr.bin; setenv m4addr 0x007F8000'
+   setenv bootm4tcml 'run m4tcml && run bootm4'
+   run bootm4tcml
+   # TCMU
+   setenv m4tcmu 'setenv m4fw zephyr.bin; setenv m4addr 0x00800000'
+   setenv bootm4tcmu 'run m4tcmu && run bootm4'
+   run bootm4tcmu
+   # OCRAM
+   setenv m4ocram 'setenv m4fw zephyr.bin; setenv m4addr 0x00900000'
+   setenv bootm4ocram 'run m4ocram && run bootm4'
+   run bootm4ocram
+   # OCRAM_S
+   setenv m4ocrams 'setenv m4fw zephyr.bin; setenv m4addr 0x00180000'
+   setenv bootm4ocrams 'run m4ocrams && run bootm4'
+   run bootm4ocrams
+   # DDR
+   setenv m4ddr 'setenv m4fw zephyr.bin; setenv m4addr 0x80000000'
+   setenv bootm4ddr 'run m4ddr && run bootm4'
+   run bootm4ddr
+
+
+Debugging
+=========
+
+Download and install `J-Link Tools`_ and `NXP iMX7D Connect CortexM4.JLinkScript`_.
+
+To run Zephyr Binary using J-Link create the following script in order to
+get the Program Counter and Stack Pointer from zephry.bin.
+
+get-pc-sp.sh:
+.. code-block:: console
+
+   #!/bin/sh
+
+   firmware=$1
+
+   pc=$(od -An -N 8 -t x4 $firmware | awk '{print $2;}')
+   sp=$(od -An -N 8 -t x4 $firmware | awk '{print $1;}')
+
+   echo pc=$pc
+   echo sp=$sp
+
+
+Get the SP and PC from firmware binary: ``./get-pc-sp.sh zephyr.bin``
+.. code-block:: console
+
+   pc=00900f01
+   sp=00905020
+
+Plug in the J-Link into the board and PC and run the J-Link command line tool:
+
+.. code-block:: console
+
+   /usr/bin/JLinkExe -device Cortex-M4 -if JTAG -speed 4000 -autoconnect 1 -jtagconf -1,-1 -jlinkscriptfile iMX7D_Connect_CortexM4.JLinkScript
+
+The following steps are necessary to run the zephyr.bin:
+1. Put the M4 core in reset
+2. Load the binary in the appropriate addr (TMCL, TCMU, OCRAM, OCRAM_S or DDR)
+3. Set PC (Program Counter)
+4. Set SP (Stack Pointer)
+5. Get the M4 core out of reset
+
+Issue the following commands inside J-Link commander:
+
+.. code-block:: console
+
+   w4 0x3039000C 0xAC
+   loadfile zephyr.bin,0x00900000
+   w4 0x00180000 00900f01
+   w4 0x00180004 00905020
+   w4 0x3039000C 0xAA
+
+With these mechanisms, applications for the ``warp7_m4`` board
+configuration can be built and debugged in the usual way (see
+:ref:`build_an_application` and :ref:`application_run` for more details).
+
+References
+==========
+
+- `Loading Code on Cortex-M4 from Linux for the i.MX 6SoloX and i.MX 7Dual/7Solo Application Processors`_
+- `J-Link iMX7D Instructions`_
+
+.. _WaRP7 Site:
```

Comment:
As written, only the first two of these references will be displayed in the generated HTML. If you'd like the rest of these to also be shown, add a target-notes directive here (with a blank line after it):

    .. target_note::

Reply:
Just the first two should be shown in this Reference session.

Reply:
OK